### ISSUE TITLE
defaultdict() with no factory method provided

### ISSUE THREAD

**Issue by Jeremiah-England** [astral-sh/ruff#20727]
### Summary

I just ran into a bug where someone accidentally left off passing the factory method to a default dictionary, which apparently is perfectly valid and just makes the `defaultdict` act like a regular `dict`.

```python
from collections import defaultdict

d = defaultdict()

d["key"]
# KeyError!
```

One use case for doing this is if you want to set the default factory later dynamically (`d.default_factory = list`).

But my guess is that _most_ of the time you have an empty `defaultdict()` is it not intentional.

I think that setting the default factory dynamically down the line is pretty weird and questionable, and needing to ignore a lint rule for that is OK. And if you don't want to ignore a line rule, you could use `defaultdict(None)` and make it explicit.
